packages/browser/src/plugins/in-app-plugin/index.ts

```diff
@@ -0,0 +1,156 @@
+import { Analytics } from '../../core/analytics'
+import { Context } from '../../core/context'
+import { Plugin } from '../../core/plugin'
+
+import { InAppEvents, SemanticEvents, newEvent, allEvents, gistToCIO } from './events'
+import Gist from 'customerio-gist-web'
+
+export { InAppEvents }
+
+export type InAppPluginSettings = {
+    siteId: string | undefined
+    events: EventListenerOrEventListenerObject | null | undefined
+
+    _env: string | undefined
+    _logging: boolean | undefined
+}
+
+export function InAppPlugin(
+    settings: InAppPluginSettings,
+  ): Plugin {
+
+    let _analytics: Analytics;
+    let _gistLoaded:boolean = false;
+    let _pluginLoaded:boolean = false;
+    const _eventTarget:EventTarget = new EventTarget();
+
+    function attachListeners() {
+        if(!_gistLoaded || _pluginLoaded)
+            return;
+
+        _analytics.on('reset', reset);
+
+        if(settings.events) {
+            allEvents.forEach((event) => {
+                _eventTarget.addEventListener(event, settings?.events as EventListenerOrEventListenerObject);
+            });
+            ['messageShown', 'messageDismissed', 'messageError'].forEach((event) => {
+                Gist.events.on(event, (message: any) => {
```

Comment:
Does the gist sdk have the equivalent of an "unload" function? (to deregister these events)

Reply:
We discussed this offline, Gist isn't actually using event handlers, just an array of callback functions to call. So we think this is ok.